DFReader.py

```diff
@@ -555,6 +555,103 @@ def set_message_timestamp(self, m):
+            dir = os.path.join(os.environ['LOCALAPPDATA'], '.mavproxy')
+        else:
+            dir = os.path.join(os.environ['HOME'], '.mavproxy')
+        if name is None:
+            return dir
+        return os.path.join(dir, name)
+
+    def metadata_tree(self,verbose=False):
+        '''return a map between a log message and its metadata. May return None if data is not available'''
+        # If we've already tried loading data, use it if we have it
+        # This avoid repeated attempts, when the file is not there
+        if self.tried:
+            return self.data
+        self.tried = True
+        # Get file name, based on vehicle type
+        mapping = { mavutil.mavlink.MAV_TYPE_GROUND_ROVER : "Rover",
+                    mavutil.mavlink.MAV_TYPE_FIXED_WING : "Plane",
+                    mavutil.mavlink.MAV_TYPE_QUADROTOR : "Copter",
+                    mavutil.mavlink.MAV_TYPE_HELICOPTER : "Copter",
+                    mavutil.mavlink.MAV_TYPE_ANTENNA_TRACKER : "Tracker",
+                    mavutil.mavlink.MAV_TYPE_SUBMARINE : "Sub",
+                    mavutil.mavlink.MAV_TYPE_AIRSHIP : "Blimp",
+                    }
+        if self.parent.mav_type in mapping:
+            path = self.dot_mavproxy(os.path.join("LogMessages","%s.xml" % mapping[self.parent.mav_type]))
+        else:
+            return None
+        # Does the file exist?
+        if not os.path.exists(path):
+            if verbose:
+                print("Can't find '%s'\nPlease run 'logmessage download' first" % path)
+            return None
+        # Read in the XML
+        xml = open(path,'rb').read()
+        from lxml import objectify
+        objectify.enable_recursive_str()
+        tree = objectify.fromstring(xml)
+        data = {}
+        for p in tree.logformat:
+            n = p.get('name')
+            data[n] = p
+        # Cache and return data
+        self.data = data
+        return self.data
+
+    def print_help(self,msg):
+        '''print help for a log message'''
+        data = self.metadata_tree(True)
+        if data is None or not msg in data:
+            return
+        node = data[msg]
+        # Message name and description
+        print("Log Message: %s\n%s\n" % (msg,node.description.text))
+        # Protect against replay messages which dont list their fields
+        if hasattr(node.fields,'field'):
+            # Loop through fields
+            for f in node.fields.field:
+                units = f.get('units')
+                dtype = f.get('type')
+                addstr = ''
+                if units:
+                    addstr = "[%s] " % units
+                elif 'char' in dtype:
+                    addstr = "[%s] " % dtype
+                elif hasattr(f,'enum'):
+                    addstr = "[enum] "
+                elif hasattr(f,'bitmask'):
+                    addstr = "[bitmask] "
+                print("%s %s: %s" % (f.get('name'),addstr,f.description.text))
```

Comment:
I imagine we might argue about this format a bit, but I'm not going to here.

... except, perhaps, to mention that fixed-width might be a reasonable option... but `degheading` is way too-long a string, really :-)

Reply:
Something like this....?
```
TimeUS   [μs]         : Time since system startup
DesRoll  [deg]        : vehicle desired roll
Roll     [deg]        : achieved vehicle roll
DesPitch [deg]        : vehicle desired pitch
Pitch    [deg]        : achieved vehicle pitch
DesYaw   [degheading] : vehicle desired yaw
Yaw      [degheading] : achieved vehicle yaw
ErrRP    [deg]        : lowest estimated gyro drift error
ErrYaw   [degheading] : difference between measured yaw and DCM yaw estimate
AEKF                  : active EKF type
```